var/spack/repos/builtin/packages/py-ats/package.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyAts(PythonPackage):
+    """ATS - Automated Testing System - is an open-source, Python-based tool
+    for automating the running of tests of an application across a broad range
+    of high performance computers."""
+
+    homepage = "https://github.com/LLNL/ATS"
+    git      = "https://github.com/LLNL/ATS.git"
+
+    maintainers = ['white238']
+
+    version('main', branch='main')
+    version('7.0.10', tag='7.0.10')
+    version('7.0.5.9', tag='7.0.5.9')
+    version('7.0.5.8', tag='7.0.5.8')
+    version('7.0.5.7', tag='7.0.5.7')
+    version('7.0.5.6', tag='7.0.5.6')
+    version('7.0.5.5', tag='7.0.5.5')
+    version('7.0.5.4', tag='7.0.5.4')
+    version('7.0.5.3', tag='7.0.5.3')
+    version('7.0.5.2', tag='7.0.5.2')
+    version('7.0.5.1', tag='7.0.5.1')
+    version('7.0.5', tag='7.0.5')
+    version('7.0.4', tag='7.0.4')
+
+    depends_on("py-numpy", type=('build', 'run'))
```

Comment:
I notice the code -- `ats/atsMachines/FutureMachines/flux_direct.py` -- imports `flux`.  Where is that coming from?

Reply:
The `test/HelloATS/READ.ME` file has a comment mentioning @trws who might know?

Reply:
> The `test/HelloATS/READ.ME` file has a comment mentioning @trws who might know?

I hadn't looked to see if it may be `flux-core` or `flux-shed` or elsewhere.  Pointed it out in case there may be a missing dependency.

Reply:
I'm guessing this file is just example code that doesn't get installed, but it would be good to get confirmation.

Reply:
It should be flux-core, optionally flux-shed for certain features, but that should be an optional dependency.  Flux, last I knew, was one of the possible schedulers and not a default one.  Dong would know more, though I can't seem to at him here.

Reply:
> It should be flux-core, optionally flux-shed for certain features, but that should be an optional dependency. Flux, last I knew, was one of the possible schedulers and not a default one. Dong would know more, though I can't seem to at him here.

In that case I recommend adding a `variant` and the associated dependency (and support).

Reply:
The flux capability in ATS doesn't work atm.  I assume you don't want a broken variant but would it be helpful if I added a TODO and/or a commented out variant?

Reply:
A comment could address potential future questions about this. 😁